test/server/cards/03_TWI/leaders/GeneralGrievousGeneralOfTheDroidArmies.spec.ts

```diff
@@ -0,0 +1,83 @@
+describe('General Grievous, General Of The Droid Armies', function () {
+    integration(function (contextRef) {
+        describe('General Grievous\'s leader undeployed ability', function () {
+            it('should Sentinel to a Droid unit for this phase', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['battle-droid-legion'],
+                        spaceArena: ['devastating-gunship'],
+                        leader: 'general-grievous#general-of-the-droid-armies'
+                    },
+                    player2: {
+                        groundArena: ['battlefield-marine', 'atst'],
+                    }
+                });
+
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.generalGrievous);
+                context.player1.clickPrompt('Give a Droid unit Sentinel for this phase');
+                expect(context.player1).toBeAbleToSelectExactly([context.devastatingGunship, context.battleDroidLegion]);
+                context.player1.clickCard(context.battleDroidLegion);
+
```

Comment:
Can you do me a favor - since Battle Droid tokens are kind of new it would be good to test that they work here. Would you mind adding the Droid Deployment card to this test and choose one of the generated droid tokens as the sentinel target? Makes this test a little more complicated but it would be helpful for confirming that the token units work right